gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -119,80 +116,95 @@ private static void configureSourceSet(
-            // but we can rely on this since we control GCV.
-            // Alternatively, we could tell GCV to lock this configuration, at the cost of a slightly more
-            // expensive 'unifiedClasspath' resolution during lock computation.
-            if (project.getRootProject().getPluginManager().hasPlugin("com.palantir.versions-lock")) {
-                explicitCompile.extendsFrom(project.getConfigurations().getByName("lockConstraints"));
-            }
-            // Inherit the excludes from compileClasspath too (that get aggregated from all its super-configurations).
-            compileClasspath.getExcludeRules().forEach(rule -> explicitCompile.exclude(excludeRuleAsMap(rule)));
-        });
+                    conf.withDependencies(deps -> {
+                        // Pick up GCV locks. We're making an internal assumption that this configuration exists,
+                        // but we can rely on this since we control GCV.
+                        // Alternatively, we could tell GCV to lock this configuration, at the cost of a slightly more
+                        // expensive 'unifiedClasspath' resolution during lock computation.
+                        if (project.getRootProject().getPluginManager().hasPlugin("com.palantir.versions-lock")) {
+                            conf.extendsFrom(project.getConfigurations().getByName("lockConstraints"));
+                        }
+                        // Inherit the excludes from compileClasspath too (that get aggregated from all its
+                        // super-configurations).
+                        compileClasspath.get().getExcludeRules().forEach(rule -> conf.exclude(excludeRuleAsMap(rule)));
+                    });
 
-        // Since we are copying configurations before resolving 'explicitCompile', make double sure that it's not
-        // being resolved (or dependencies realized via `.getIncoming().getDependencies()`) too early.
-        AtomicBoolean projectsEvaluated = new AtomicBoolean();
-        project.getGradle().projectsEvaluated(g -> projectsEvaluated.set(true));
-        explicitCompile
-                .getIncoming()
-                .beforeResolve(ir -> Preconditions.checkState(
-                        projectsEvaluated.get()
-                                || (project.getGradle().getStartParameter().isConfigureOnDemand()
-                                        && project.getState().getExecuted()),
-                        "Tried to resolve %s too early.",
-                        explicitCompile));
+                    // Since we are copying configurations before resolving 'explicitCompile', make double sure that
+                    // it's not
+                    // being resolved (or dependencies realized via `.getIncoming().getDependencies()`) too early.
+                    AtomicBoolean projectsEvaluated = new AtomicBoolean();
+                    project.getGradle().projectsEvaluated(g -> projectsEvaluated.set(true));
```

Comment:
an aside, do you want some sort of similar check in the flattener?

Reply:
Honestly, it's [a bit redundant given GCV does the same thing](https://github.com/palantir/gradle-consistent-versions/blob/f407769c01dba863147e6108bd12e6990ab2ebed/src/main/java/com/palantir/gradle/versions/VersionsPropsPlugin.java#L220-L230)